services/src/main/java/org/exoplatform/gamification/rest/UserReputationEndpoint.java

```diff
@@ -0,0 +1,291 @@
+package org.exoplatform.gamification.rest;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.exoplatform.commons.file.model.FileItem;
+import org.exoplatform.commons.file.services.FileService;
+import org.exoplatform.commons.file.services.FileStorageException;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.gamification.entities.domain.effective.GamificationContextItemEntity;
+import org.exoplatform.gamification.service.configuration.BadgeService;
+import org.exoplatform.gamification.service.dto.configuration.BadgeDTO;
+import org.exoplatform.gamification.service.effective.GamificationService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.services.security.ConversationState;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.service.rest.Util;
+import org.exoplatform.social.webui.Utils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Path("/gamification/reputation")
+@Produces(MediaType.APPLICATION_JSON)
+public class UserReputationEndpoint implements ResourceContainer {
+
+    private static final Log LOG = ExoLogger.getLogger(UserReputationEndpoint.class);
+
+    private final CacheControl cacheControl;
+
+    protected GamificationService gamificationService = null;
+
+    protected IdentityManager identityManager = null;
+
+    protected BadgeService badgeService = null;
+
+    public UserReputationEndpoint() {
+
+        this.cacheControl = new CacheControl();
+
+        cacheControl.setNoCache(true);
+
+        cacheControl.setNoStore(true);
+
```

Comment:
We should think about the cache control on user methods

Reply:
In leaderboard app we should not use caches since social actions may change user's score in every minute